README.md
## MoneyTigo Payment Module for PrestaShop 1.6 & 1.7

This payment module allows you to accept credit card payments through MoneyTigo.com payment solution (https://www.moneytigo.com).
This payment module is compatible with all versions of prestashop 1.6 & 1.7

* Module version: 1.1.1

# INSTALLATION FOR THE FIRST TIME

To install MoneyTigo plugin we invite you first : 

### For prestashop V1.6

* As a priority - [Download the archive of the module by clicking here](https://github.com/moneytigo/prestashop_moneytigo/releases/download/v1.1.2/moneytigo-1617-v_1_1_2.zip)
* 1 - Go to your PRESTASHOP dashboard
* 2 - Go to the module manager (modules and services > modules and services)
* 3 - At the top right, click on "Add a new module" then "select the Moneytigo module archive" then click on "Load the module".
* 4 - The module is loaded, you must click on the button "install" then "continue installation".
* 5 - Fill in your API key and your SECRET key (which you will find in your MoneyTigo space) then click on "Update configuration".
* 6 - Then go to (modules and services > payment) in the section "country restriction" check the countries in which you want to display MoneyTigo then save
* 7 - The installation is finished and functional!

(**Note:** To test transactions, don't forget to switch your website (in your MoneyTigo interface, in test/demo mode) and switch it to production when your tests are finished.

If you use the test mode you must use the following virtual credit cards:
* **Payment approved** : Card n° 4000 0000 0000 0002 , Expiry 12/22 , Cvv 123
* **Payment declined** : Card n° 4000 0000 0000 0036 , Expiry 12/22, Cvv 123
* **(Virtual cards do not work in production mode)**

### For prestashop V1.7

* As a priority - [Download the archive of the module by clicking here](https://github.com/moneytigo/prestashop_moneytigo/releases/download/v1.1.2/moneytigo-1617-v_1_1_2.zip)
* 1 - Go to your PRESTASHOP dashboard
* 2 - Go to the module manager (modules > module managers)
* 3 - At the top right, click on "Install a module" then "select the Moneytigo module archive".
* 4 - The module is loaded, you must click on the button "Configure".
* 5 - Fill in your API key and your SECRET key (which you will find in your MoneyTigo space) then click on "Update configuration".
* 6 - Then go to (payment > preferences) in the section "country restrictions" check the countries in which you want to display MoneyTigo then save
* 7 - The installation is finished and functional!

(**Note:** To test transactions, don't forget to switch your website (in your MoneyTigo interface, in test/demo mode) and switch it to production when your tests are finished.

If you use the test mode you must use the following virtual credit cards:
* **Payment approved** : Card n° 4000 0000 0000 0002 , Expiry 12/22 , Cvv 123
* **Payment declined** : Card n° 4000 0000 0000 0036 , Expiry 12/22, Cvv 123
* **(Virtual cards do not work in production mode)**

# UPDATE OF AN OLD MONEYTIGO MODULE

The procedure of update is the same in both version of prestashop you just have to follow the points 1 to 4 indicated above, prestashop will automatically update the module with the new version and keeping all your data so it will not be necessary to reconfigure it, if the module is not displayed after update we just invite you to check the prestashop restrictions mentioned in the point 6.


changelog.md
2021-12-14 - Updating
Fix bug during installation on prestashop 1.7.8.2 version

2021-07-05 - Updating
Optimization of the payment process
Adding the payment in 2 times
Added payment in 4 times
Added fee rules for multiple payment transactions

2021-02-19 - Updating
Change of name of the IPS Payment payment solution to MoneyTigo.
Improvement of the ordering process
Setting up payment by iframe and keeping the customer on the merchant's website.

2020-11-14 - Updating
Update of the module to integrate the latest modifications of the iPS Payment API as well as the latest security developments.

2019 MONEYTIGO

NOTICE OF LICENSE

This module was developed by IPS Internationnal for the use of the IPS-PAYMENT payment service:
simple and secure online payment solution.
For any questions relating to its use, please contact us
at dev@ips-payment.com

controllers/front/payment.php
<?php
/**
    * NOTICE OF LICENSE
    *
    * This file is create by Ips
    * For the installation of the software in your application
    * You accept the licence agreement.
    *
    * You must not modify, adapt or create derivative works of this source code
    *
    *  @author    Ips
    *  @copyright 2018-2021 IPS INTERNATIONNAL SAS
    *  @license   moneytigo.com
*/

class MoneytigoPaymentModuleFrontController extends ModuleFrontController
{
    public function initContent()
    {
        $this->display_column_left = false;
        parent::initContent();
        $context = $this->context;
        $cart = $context->cart;
        $customer = new Customer((int)$cart->id_customer);
        $extension = new Moneytigo();
        if (!$extension->isAvailable()) {
            Tools::redirect('index.php?controller=order');
        }
        if ((Tools::getvalue('typeTr') == 1) && (Configuration::get('MONEYTIGO_GATEWAY_P3F')=="on")) {
            $P3F= "P3F";
            $context->cookie->__set("typeTr", $P3F);
        } else {
            $P3F = "D";
            $context->cookie->__set("typeTr", $P3F);
        }

        
        //construct payment request
        
        $moneytigo = array();
        $moneytigo['amount'] = number_format($cart->getOrderTotal(), 2, '.', '');

        if ((Tools::getvalue('typeTr') == 1) && (Configuration::get('MONEYTIGO_GATEWAY_P3F')=="on")) {
            $moneytigo['Lease'] = "3"; //Its payment in three time
            $P3F= "P3F";
            $context->cookie->__set("typeTr", $P3F);
        } else {
            $P3F = "D";
            $context->cookie->__set("typeTr", $P3F);
        }
        $moneytigo['RefOrder'] = (int)$cart->id."-".$P3F;
        $moneytigo['lang'] = Language::getIsoById($this->context->language->id);
        $getsUrlRetour = $this->context->link->getModuleLink(
            'moneytigo',
            'return',
            array('customer'=>$customer->secure_key, 'id_cart'=>(int)$cart->id)
        );
        $moneytigo['urlOK'] = $getsUrlRetour;
        $moneytigo['urlKO'] = $getsUrlRetour;
        $urlIPN = (Configuration::get('PS_SSL_ENABLED') ? 'https' : 'http');
        $urlIPN .= '://'.$_SERVER['HTTP_HOST'].__PS_BASE_URI__.'modules/'.$this->module->name.'/ipn.php';
        $moneytigo['urlIPN'] = $urlIPN;
        $moneytigo['extension'] = "prestashop-".$this->module->name."-".$this->module->version;
        $moneytigo['email_Customer'] = $customer->email;
        $moneytigo['name_Customer'] = $customer->lastname;
        $moneytigo['lastname_Customer'] = $customer->firstname;
        $moneytigo['MerchantKey'] = Configuration::get('MONEYTIGO_GATEWAY_API_KEY'); //Api Key identify Merchant !
        $form_action = Configuration::get('MONEYTIGO_URI_API_PAYMENT');
        // Datas Form
        $arraySmarty = array(
            'isCartEmpty' => ((int)$cart->getOrderTotal() == 0) ? true:false,
            'form_action' => $form_action
        );
        foreach ($moneytigo as $key => $value) {
            $arraySmarty[$key] = $value;
        }
        
      
		
        $this->context->smarty->assign($arraySmarty);
        if (_PS_VERSION_ < '1.7') {
            $this->setTemplate('16/moneytigoold.tpl'); //1.6
        } else {
            $this->setTemplate('module:moneytigo/views/templates/front/17/moneytigo.tpl'); //1.7
        }
    }
}
